contentcuration/contentcuration/frontend/shared/vuex/indexedDBPlugin/index.js

```diff
@@ -0,0 +1,161 @@
+import { EventEmitter } from 'events';
+import { CHANGE_TYPES } from 'shared/data';
+import { CLIENTID } from 'shared/data/db';
+
+function getEventName(table, type) {
+  return `${table}/${type}`;
+}
+
+export class Listener {
+  /**
+   * @param {Function} callback
+   */
+  constructor(callback) {
+    this.callback = callback;
+    this.tableName = null;
+    this.changeType = null;
+    this.namespacePrefix = null;
+  }
+
+  /**
+   * @return {string|null}
+   */
+  getEventName() {
+    if (!this.tableName || !this.changeType) {
+      return null;
+    }
+    return getEventName(this.tableName, this.changeType);
+  }
+
+  /**
+   * @param {string} name
+   * @return {string}
+   */
+  prefix(name) {
+    return this.namespacePrefix ? `${this.namespacePrefix}/${name}` : name;
+  }
+
+  /**
+   * @param {String} tableName
+   * @param {String|Number} changeType
+   * @param {String|null} [namespacePrefix]
+   * @return {Listener}
+   */
+  bind(tableName, changeType, namespacePrefix = null) {
+    changeType = Number(changeType);
+    if (!Object.values(CHANGE_TYPES).includes(changeType)) {
+      throw RangeError(
+        `Change must be ${CHANGE_TYPES.CREATED}, ${CHANGE_TYPES.UPDATED}, or ${CHANGE_TYPES.DELETED}`
+      );
+    }
+
+    const listener = new this.constructor(this.callback);
+    listener.tableName = tableName;
+    listener.changeType = changeType;
+    listener.namespacePrefix = namespacePrefix;
+    return listener;
+  }
+
+  /**
+   * @param {EventEmitter} events
+   * @param {Store} store
+   */
+  register(events, store) {
+    const eventName = this.getEventName();
+
+    if (!eventName) {
+      console.warn('Cannot register unbound listener: ' + this.callback.toString());
+      return;
+    }
+
+    events.addListener(eventName, obj => {
+      this.callback(store, obj);
+    });
+  }
+}
```

Comment:
Listener class represents a callback, which can be bound to a specific table and change type.